applications/tari_validator_node/src/payload_processor.rs

```diff
@@ -85,26 +89,35 @@ where TTemplateProvider: TemplateProvider<Template = LoadedTemplate>
             initial_ownership_proofs: vec![owner_token],
         };
 
-        let modules = vec![]; // No modules for now, currently used in tests. Also will be useful for more advanced use-cases like fees, etc.
-
-        let processor = TransactionProcessor::new(package, state_store, auth_params, consensus, modules);
+        // 1 per byte
+        // Divide by 2 to account for the cost of CBOR
+        let initial_cost = self.fee_table.per_kb_wasm_size() * package.total_code_byte_size() as u64 / 1024 / 2;
```

Comment:
nit: this logic is probably better in a separate struct or service, but fine to leave it here until more rules emerge

Reply:
yeah agree, might want to charge for args size etc so possibly just a cost function that takes in a Transaction will be sufficient.